src/native/eventpipe/ep-stack-contents.h

```diff
@@ -21,21 +21,21 @@ struct _EventPipeStackContents {
 #else
 struct _EventPipeStackContents_Internal {
```

Comment:
I think you should do a "zero length array" instead of keeping EP_MAX_STACK_DEPTH of the struct:

uintptr_t stack_frames [1];

and then just update size calculation based on number of stack frames as described by next_available_frame.

For the methods, we could on checked build just append similar as done now and just adjust when calculating size. You could also pretty much keep all of the getters and setters, just make sure they calculate offset to "stack_frames" and "methods" since they will both be based out of new zero length array. Maybe we should rename it to "data" instead since it will be an array of uintptr_t to next_available_frame - 1 stack frames followed by (on checked build) next_available_frame -1 methods.

Reply:
To be clear, are you suggesting we remove the `EventPipeStackContents` field from `EventPipeEventInstance` and replace it with `uint32_t stack_data_len; uintptr_t *stack_data;`? It would simplify some of the logic where I just subtract out `sizeof (EventPipeStackInstance)`.

I'll play with that. It makes sense, since we are mutating the fixed size struct into a dynamic/flattened format just like the `EventPipeEvent`->`EventPipeEventInstance` conversion.

Reply:
Yes, something along the lines, we will still need a fixed size EventPipeStackContents since it is used to capture the stack and passed into write event, but not need for EventPipeEventInstance, so that could use the "zero length" array and it should be possible to get the stack frame location as well as methods calculated out from that, so most of the getter/setters should remain intact and it should be simpler to calculate needed size of a new EventPipeEventInstance as well as the size of an existing EventPipeEventInstance.

NOTE, maybe we should take opportunity to raise the number of stack slots in EventPipeStackContents as well (it is only stack allocated), we have customers hitting the limit where callstacks surpass 100 frames.